RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.cc

```diff
@@ -0,0 +1,113 @@
+#include "RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/Utilities/interface/do_nothing_deleter.h"
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+using namespace edm;
+
+OnlineBeamSpotESProducer::OnlineBeamSpotESProducer(const edm::ParameterSet& p) {
+  auto cc = setWhatProduced(this);
+
+  transientBS_ = new BeamSpotObjects;
```

Comment:
appears to leak as transientBS gets assigned to theHLTBS_ or theLegacyBS_ below without transisentBS_ being deleted. I would set this to a null ptr  as I dont see the need for creating this object